.github/workflows/build-all.yml

```diff
@@ -171,25 +188,44 @@ jobs:
         run: |
           sleep 20
           OUTPUT="$(docker ps --filter "name=node" --format "{{.Status}}" | grep -o 'Up [0-9]\+ seconds' | cut -d' ' -f2 | awk -v ORS='\\n' '1' | head -c -2)"
-          echo "::set-output name=count::$({ echo -e $OUTPUT | wc -l ; echo -e $OUTPUT | awk -F= '$1<30' | wc -l; } | sort -n | head -1)"
+          echo "count=$({ echo -e $OUTPUT | wc -l ; echo -e $OUTPUT | awk -F= '$1<30' | wc -l; } | sort -n | head -1)" >> $GITHUB_OUTPUT
       - name: Fail if not all nodes restarted      
         if: steps.nodes_restarted.outputs.count != 4
         uses: actions/github-script@v3
         with:
           script: core.setFailed('${{ matrix.os_version }} - Not all nodes have been restarted')
 
+  statusCheck:
+    name: statusCheck
+    runs-on: ubuntu-latest
+    needs: [workflow_setup, test_node_and_controller]
+    if: ${{ needs.workflow_setup.outputs.testsNeeded == 'false' || success() }}
+    steps:
+      - run: 'echo "Just a status Check (Always true, when executed) for branch protection rules(blocks merging while test are running and if tests fail)." '
+
   push_node:
     runs-on: ubuntu-latest
     needs: [workflow_setup, test_node_and_controller]
+    if: ${{ github.event_name != 'pull_request' }}
     strategy:
       matrix: ${{ fromJson(needs.workflow_setup.outputs.os_matrix) }}
     steps:
       - name: Check out the repo
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+      - name: indy-node-version
+        id: indy-node-version
+        shell: bash
+        run: |
+          export nodeVersion=$(sed "s/~/-/" <<<$(grep -oP "indy-node=\"?\d+.\d+.\d+((~|.)?rc\d+)?\"?" build/Dockerfile.ubuntu20 | grep -oP "\d+.\d+.\d+((~|.|-)?rc\d+)?"))
```

Comment:
grep needs to look into the respective docker files for each build. Not always ubuntu20 ;)

Reply:
I dont know if this will work in terms af variable replacement: 
```suggestion
          export nodeVersion=$(sed "s/~/-/" <<<$(grep -oP "indy-node=\"?\d+.\d+.\d+((~|.)?rc\d+)?\"?" build/Dockerfile.-${{ matrix.os_version }} | grep -oP "\d+.\d+.\d+((~|.|-)?rc\d+)?"))
```

Reply:
The `-` was too much but it works:

![image](https://user-images.githubusercontent.com/87540518/201636696-6952b33f-7c3e-4ea0-b4e4-5078ce30ec82.png)